src/Plugin/Plugins/TempHumidPlugin.cpp

```diff
@@ -0,0 +1,259 @@
+    if (nullptr == m_slotInterf) {
+        m_pageTime = 10000U;
+    } else {
+        m_pageTime = m_slotInterf->getDuration() / 4;
+    }
+
+    gfx.fillScreen(ColorDef::BLACK);
+
+    if (nullptr == m_iconCanvas)
+    {
+        m_iconCanvas = new Canvas(ICON_WIDTH, ICON_HEIGHT, 0, 0);
+
+        if (nullptr != m_iconCanvas)
+        {
+            (void)m_iconCanvas->addWidget(m_bitmapWidget);
+
+            /* Load icon from filesystem. */
+            (void)m_bitmapWidget.load(FILESYSTEM, IMAGE_PATH_TEMP_ICON);
+
+            m_iconCanvas->update(gfx);
+        }
+    }
+    else
+    {
+        m_iconCanvas->update(gfx);
+    }
+
+    if (nullptr == m_textCanvas)
+    {
+        m_textCanvas = new Canvas(gfx.getWidth() - ICON_WIDTH, gfx.getHeight(), ICON_WIDTH, 0);
+
+        if (nullptr != m_textCanvas)
+        {
+            (void)m_textCanvas->addWidget(m_textWidget);
+
+            m_textCanvas->update(gfx);
+        }
+    }
+    else
+    {
+        m_textCanvas->update(gfx);
+    }
+
+    unlock();
+
+    return;
+}
+
+void TempHumidPlugin::update(IGfx& gfx)
+{
+    bool            showPage    = false;
+    char            tmp[6]      = "";     // holds the display value
+
+    if (false == m_timer.isTimerRunning())
+    {
+        m_timer.start(m_pageTime);
+        showPage = true;
+    }
+    else if (true == m_timer.isTimeout())
+    {
+        ++m_page;
+        m_page %= 2U;
+
+        showPage = true;
+        m_timer.restart();
+    }
+    else
+    {
+        ;
+    }
+
+    if (true == showPage)
+    {
+        /* Clear display */
+        gfx.fillScreen(ColorDef::BLACK);
+
+        if (nullptr != m_iconCanvas)
+        {
+            m_iconCanvas->update(gfx);
+        }
+
+        if (nullptr != m_textCanvas) 
+        {
+            m_textCanvas->update(gfx);
+        }
+
+        switch(m_page)
+        {
+        case 0U: // screen 0 is temperature
+            m_bitmapWidget.load(FILESYSTEM, IMAGE_PATH_TEMP_ICON);
+            snprintf(tmp, sizeof(tmp), "%3f", m_temp);
+            m_text = tmp;
+            //somehow the degree symbol does not get displayed and the text gets truncated! m_text += "\x8E";
+            m_text += " C";
+            LOG_INFO("p0: new string %s", m_text);
+            m_textWidget.setFormatStr(m_text);
+            break;
+
+        case 1U: // screen 1 is humidity
+            m_bitmapWidget.load(FILESYSTEM, IMAGE_PATH_HUMID_ICON);
+            snprintf(tmp, sizeof(tmp), "%3f", m_humid);
+            m_text = tmp;
+            m_text += "%";
+            LOG_INFO("p1: new string %s", m_text);
+            m_textWidget.setFormatStr(m_text);
+            break;
+
+        default:
+            break;
+        }
+    }
+
+    return;
+}
+
+void TempHumidPlugin::process() 
+{
+    unsigned long m=millis();
+    float h;
+    float t;
+
+   /* read only if update_period not reached or sensor has never been read */
+   if ( m_last + SENSOR_UPDATE_PERIOD < m  || m_last == 0)
```

Comment:
Please ensure with braces what shall be done first. The preincidence is not clear always.